leaders/src/main/java/edu/cornell/worker/DockerWorker.java

```diff
@@ -0,0 +1,159 @@
+package edu.cornell.worker;
+
+import static edu.cornell.Main.ENV_KAFKA_ADDRESS;
+import static edu.cornell.Main.ENV_REPO_BRANCH;
+import static edu.cornell.Main.ENV_REPO_TESTS;
+import static edu.cornell.Main.ENV_REPO_URL;
+
+import com.github.dockerjava.api.DockerClient;
+import com.github.dockerjava.api.async.ResultCallback;
+import com.github.dockerjava.core.DefaultDockerClientConfig;
+import com.github.dockerjava.core.DockerClientConfig;
+import com.github.dockerjava.core.DockerClientImpl;
+import com.github.dockerjava.httpclient5.ApacheDockerHttpClient;
+import com.github.dockerjava.transport.DockerHttpClient;
+import java.io.Closeable;
+import java.io.IOException;
+import java.time.Duration;
+import java.util.List;
+import java.util.Set;
+import lombok.EqualsAndHashCode;
+import lombok.NonNull;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+
+/** A worker that runs inside a Docker container */
+@Slf4j
+@EqualsAndHashCode
+@ToString
+public class DockerWorker implements Worker {
+
+    /**
+     * The link to the workers Docker image
+     */
+    private static final String IMAGE_URL = "ghcr.io/jakejack13/speed-workers:latest";
+
+    /**
+     * The Docker http client
+     */
+    private final DockerHttpClient httpClient;
+
+    /**
+     * The Docker client
+     */
+    private final DockerClient dockerClient;
+    /**
+     * The id of the container running the worker
+     */
+    private final String id;
+
+    /**
+     * true if the worker has finished running, false otherwise
+     */
+    private boolean done = false;
+
+    /**
+     * Creates a new DockerWorker
+     * @param repoUrl the url of the repository
+     * @param repoBranch the branch of the repository to test
+     * @param tests the set of tests to run
+     * @param kafkaAddress the address of the Kafka message bus to send test results to
+     */
+    DockerWorker(@NonNull String repoUrl, @NonNull String repoBranch, @NonNull Set<String> tests,
+            @NonNull String kafkaAddress) {
+        DockerClientConfig config =
+                DefaultDockerClientConfig.createDefaultConfigBuilder()
+                        .build();
+        httpClient = new ApacheDockerHttpClient.Builder()
+                .dockerHost(config.getDockerHost())
+                .sslConfig(config.getSSLConfig())
+                .maxConnections(100)
+                .connectionTimeout(Duration.ofSeconds(30))
+                .responseTimeout(Duration.ofSeconds(45))
+                .build();
+        dockerClient = DockerClientImpl.getInstance(config, httpClient);
+        try {
+            dockerClient.pullImageCmd(IMAGE_URL).start().awaitCompletion();
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+        id = dockerClient.createContainerCmd(IMAGE_URL)
+                .withPortSpecs("29092:29092")
```

Comment:
Worker seems to run just fine on the host computer but as soon as it's in a Docker container, it can't connect to Kafka. I tried doing this port forwarding but doesn't seem to work. Will need to look into it more. 

Things to check
- [ ] Create Docker network via docker-compose

Reply:
Seems to work when local Kafka broker is set to 29092 instead of 9092

Reply:
Fixed